src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -93,8 +93,9 @@ public interface IChainedHeaderTree
         /// All peers are checked against max reorg violation and if they violate their chain will be reset.
         /// </remarks>
         /// <param name="newConsensusTip">The new consensus tip.</param>
+        /// <param name="blockMined">Was the block mined or received from the network.</param>
         /// <returns>List of peer Ids that violate max reorg rule.</returns>
-        List<int> ConsensusTipChanged(ChainedHeader newConsensusTip);
+        List<int> ConsensusTipChanged(ChainedHeader newConsensusTip, bool blockMined = false);
```

Comment:
I prefer you made changes to tree and CM in a separate PR and not mix it with POA stuff.